cookbook/src/tx_segwit-v0.md

```diff
@@ -0,0 +1,342 @@
+
+We compute the signature `sig` by using the [`sign_ecdsa`](https://docs.rs/secp256k1/latest/secp256k1/struct.Secp256k1.html#method.sign_ecdsa) method.
+It takes a refence to a [`Message`](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html) and a reference to a [`SecretKey`](https://docs.rs/secp256k1/latest/secp256k1/struct.SecretKey.html) as arguments,
+and returns a [`Signature`](https://docs.rs/secp256k1/latest/secp256k1/ecdsa/struct.Signature.html) type.
+
+In the next step, we update the `sighash_cache` first with [`witness_mut`](https://docs.rs/bitcoin/latest/bitcoin/sighash/struct.SighashCache.html#method.witness_mut) method.
+This allows us to modify the witness stack of the input we are signing.
+It returns an [`Option<&mut Witness>`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/witness/struct.Witness.html) type.
+Next, the witness stack is by using the [`push_bitcoin_signature`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/witness/struct.Witness.html#method.push_bitcoin_signature) method.
+It expects two arguments:
+
+1. A reference to a [`SerializedSignature`](https://docs.rs/secp256k1/0.27.0/secp256k1/ecdsa/serialized_signature/struct.SerializedSignature.html) type.
+   This is accomplished by calling the [`serialize_der`](https://docs.rs/secp256k1/latest/secp256k1/ecdsa/struct.Signature.html#method.serialize_der) method on the `Signature` `sig`,
+   which returns a `SerializedSignature` type.
+1. A [`EcdsaSighashType`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html) enum.
+   Again we are using the same [`All`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html#variant.All) variant that we used earlier.
+
+Finally, we convert the `sighash_cache` back to a [`Transaction`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/transaction/struct.Transaction.html) type by using the [`into_transaction`](https://docs.rs/bitcoin/latest/bitcoin/sighash/struct.SighashCache.html#method.into_transaction) method.
+
+As the last step we print this to terminal using the [`println!`](https://doc.rust-lang.org/std/macro.println.html) macro.
+This transaction is now ready to be broadcasted to the Bitcoin network.
+
+[^today]: mid-2023.
+
+[^change]: Please note that the `CHANGE_AMOUNT` is not the same as the `DUMMY_UTXO_AMOUNT` minus the `SPEND_AMOUNT`.
+           This is due to the fact that we need to pay a fee for the transaction.
+
+[^expect]: We will be unwraping any [`Option<T>`](https://doc.rust-lang.org/std/option)/[`Result<T, E>`](https://doc.rust-lang.org/std/result) with the `expect` method.
+
+[^ThirtyTwoByteHash]: Additionally it is also necessary that the hash is 32-bytes long.
```

Comment:
If you decide to use my suggestion then this goes away. I'm not sure we should bother folk at this level with info on this trait.